src/irrlicht_changes/CGUITTFont.h

```diff
@@ -32,23 +33,52 @@
 
 #pragma once
 
-#include <map>
 #include <ft2build.h>
-#include FT_FREETYPE_H
+#include <freetype/freetype.h>
 
 #include "IGUIEnvironment.h"
 #include "IGUIFont.h"
 #include "IVideoDriver.h"
 #include "IrrlichtDevice.h"
-#include "SMesh.h"
 #include "util/enriched_string.h"
 #include "util/basic_macros.h"
 
+#include <map>
+#include <optional>
+
 namespace irr
 {
 namespace gui
 {
-	struct SGUITTFace;
+	// Manages the FT_Face cache.
+	struct SGUITTFace : public irr::IReferenceCounted
+	{
+	private:
+
+		static std::map<io::path, SGUITTFace*> faces;
+		static std::optional<FT_Library> freetype_library;
```

Comment:
`FT_Library` is a pointer so this should be perfectly fine without optional

Reply:
the original code had a bool and a `FT_Library` static var, which I just turned into an optional. but indeed this is a pointer and the code is clearer if we take advantage of that.